kyo-core/shared/src/test/scala/kyoTest/typeMapTest.scala

```diff
@@ -157,14 +158,44 @@ class typeMapTest extends KyoTest:
     }
 
     ".prune" - {
-        "Env[Int & String] -> Env[Int]" in pendingUntilFixed {
-            assertCompiles(
-                """
-                  | val e = TypeMap(42, "")
-                  | val p = e.prune[Int]
-                  | assert(p.size == 1)
-                  |""".stripMargin
-            )
+        "Env[Int] -> Env[Any]" in {
+            val p = TypeMap(42).prune[Any]
+            assert(p.size == 1)
+        }
+        "Env[Int & String] -> Env[Int]" in {
+            val e = TypeMap(42, "")
+            val p = e.prune[Int]
+            assert(p.size == 1)
+        }
+        "Env[Sub] -> Env[Super]" in {
+            val e = TypeMap(new RuntimeException)
+            val p = e.prune[Exception].prune[Throwable]
+            assert(p.size == 1)
+        }
+        "Env[Super] -> Env[Sub]" in {
+            assertDoesNotCompile("""
+                  | val e = TypeMap(new Throwable)
+                  | val p = e.prune[Exception]
+                  |""".stripMargin)
+        }
+        "complex" in {
+            val e =
+                TypeMap
+                    .empty
+                    .add(new Throwable)
+                    .add(new Exception)
+                    .add(new RuntimeException)
+                    .add(new NullPointerException)
+                    .add(new ClassCastException)
+            val p = e.prune[RuntimeException]
+            assert(p.size == 3)
```

Comment:
In this case no - these are just types which don't require custom classes that show the behaviors I expect.

In general, Kyo tags will support intersections soon and then it will be useful to retain multiple.